server/tests/user.signin.test.js

```diff
@@ -0,0 +1,143 @@
+/* eslint-disable import/no-extraneous-dependencies */
+// import regeneratorRuntime from 'regenerator-runtime';
+import chai from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../app';
+
+chai.use(chaiHttp);
+const { expect } = chai;
+
+describe('~~~ Testing Route (User Sign-In): POST /api/v1/users/signin ~~~', () => {
+  // User exists, and is verified
+  it('should authenticate an existing verified user', (done) => {
+    const user = {
+      firstname: 'Test',
+      lastname: 'User',
+      email: 'test@example.com',
+      password: 'test',
+      bio: 'I am an example user account. Thank you.',
+    };
+
+    // First register a sample user
+    chai
+      .request(app)
+      .post('/api/v1/users/signup')
+      .send(user)
+      .end((err, res) => {
+        expect(res).to.have.status(201);
+        // Verify users manually
+        const token = process.env.SIGNUP_TEST_TOKEN;
+        chai
+          .request(app)
+          .get(`/api/v1/users/verify/${token}`)
+          .end((err, res) => {
+            expect(res).to.have.status(200);
```

Comment:
@fordarnold Keep the test cases separate. This file is supposed to contain tests of the sign-in functionality NOT signup.